src/main/java/org/opensearch/flowframework/rest/RestCreateWorkflowAction.java

```diff
@@ -90,9 +90,9 @@ protected RestChannelConsumer prepareRequest(RestRequest request, NodeClient cli
 
             String workflowId = request.param(WORKFLOW_ID);
             Template template = Template.parse(request.content().utf8ToString());
-            boolean dryRun = request.paramAsBoolean(DRY_RUN, false);
+            boolean provision = request.paramAsBoolean(PROVISION_WORKFLOW, false);
```

Comment:
Can we keep both `dryRun` and `provision`?

if `dryrun=true`, then we validate and then index into the global context,

if `provision=true`, then we validate and then index into the global context, and then provision

Reply:
I agree we should keep both but we should not allow both provision and dryRun to be true at the same time.

Reply:
I agree we can have check here to throw an exception if they are both equaling true. 

The naming here is a little weird because when I see a `dryrun` or `validate` I don't expect to be creating something. And when I create something I would assume some inherit validation is done. However in our case users might want to create even with errors and come back to edit the template later (its more of a save action).

I think for now, like Dan said we should not allow both provision and dryrun to be true together but we should revisit this in the issue Jackie made, I will put further thoughts there